conans/client/rest/rest_client.py

```diff
@@ -144,7 +144,7 @@ def get_package_info(self, package_reference):
         contents = {key: decode_text(value) for key, value in dict(contents).items()}
         return ConanInfo.loads(contents[CONANINFO])
 
-    def get_recipe(self, conan_reference, dest_folder):
+    def get_recipe(self, conan_reference, dest_folder, retrieve_sources=False):
```

Comment:
What about a parameter being a "filtering lambda"? This way we could event avoid the CONANFILE existence.

We could extract all this code:

```
        if CONANFILE not in list(urls.keys()):
            raise NotFoundException("Conan '%s' doesn't have a %s!" % (conan_reference, CONANFILE))

        # If a filename is specified, only retrieve that file
        # if not existing, return None, no need to try to download
        if retrieve_sources:
            file_url = urls.get(EXPORT_SOURCES_TGZ_NAME)
            if file_url:
                urls = {EXPORT_SOURCES_TGZ_NAME: file_url}
            else:
                return None
        else:
            urls.pop(EXPORT_SOURCES_TGZ_NAME, None)
```